data/routesPosts.js

```diff
@@ -0,0 +1,100 @@
+const express = require('express');
+
+const PostDB = require('./helpers/postDb.js');
+
+const routesPosts = express.Router();
+
+
+// Server.js handles the /api path. Router.js handles the posts path
+
+
+// Get all the posts
+routesPosts.get('/', async (req, res) => {
+    try {
+        const posts = await PostDB.get(req.query);
+        res.status(200).json(posts);
+    } catch (error) {
+        res.status(500).json({ error: "The posts could not be retrieved." });
+    }
+});
+
+
+// Get single post by id
+routesPosts.get('/:id', async (req, res) => {
+    try {
+        const postById = await PostDB.getById(req.params.id);
+        if (postById.length === 0) {  
+            res.status(404).json({ message: "The post with the specified ID does not exist." });
+        } else {
+            res.status(200).json(postById);
+        }
+    } 
+    catch (error) { 
+        res.status(500).json({ error: "The post could not be retrieved." });
+    }
+});
+
+
+// Add a new post as user by id
+routesPosts.post('/', async (req, res) => {
+    if (!req.body.text) {
+        return res.status(409).json({ message: "Post cannot be blank!" });
+    }
+    if (!req.body.user_id) {
+        return res.status(409).json({ message: "Please include user id." });
```

Comment:
hmm, `conflict` status--could be interpreted as correct, but since it should be the user submitting this request I'd say more directly blame them with a `bad request` of 400

now that you've done this though, think about how you would account for something else: that the provided `user_id` is actually referencing a currently existing user, and not one that is either non-existent or was recently deleted. will probably involve another API call for verification but this time to the usersDB instead

Reply:
Good point about the status code. 400 is more appropriate.

Great idea about checking for existence of user. Pushed an update that includes this check. Needed to make a small adjustment to postDb.js in order to keep my router syntax nice and clean.